src/solidity/contracts.js

```diff
@@ -0,0 +1,167 @@
+    return this.decodeState(locations, storage)
+  },
+
+  /**
+   * decode the ctr state storage
+   *
+   * @param {Array} storage location  - location of all state variables
+   * @param {Map} storage  - storage
+   * @return {Map} - decoded state variable
+   */
+  decodeState: function (storageLocations, storage) {
+    if (storage['0x']) {
+      storage['0x00'] = storage['0x']
+    }
+    var ret = {}
+    for (var k in storageLocations) {
+      var stateVar = storageLocations[k]
+      var location = {
+        slot: stateVar.location.slot,
+        offset: stateVar.location.offset
+      }
+      ret[stateVar.name] = decoder.decode(stateVar.type, storage, location)
+    }
+    return ret
+  },
+
+  /**
+   * return all storage location variables of the given @arg contractName
+   *
+   * @param {String} contractName  - name of the contract
+   * @param {Object} astList  - AST nodes of all the sources
+   * @return {Object} - return the location of all vars in the storage
+   */
+  getStorageLocationOf: function (contractName, astList) {
+    var stateVars = this.getStateDefinition(astList, contractName)
+    var ret = []
+    var location = {
+      offset: 0,
+      slot: 0
+    }
+    for (var k in stateVars) {
+      var variable = stateVars[k]
+      if (variable.name === 'VariableDeclaration') {
+        var type = varUtil.getType(variable, stateVars)
+        var loc = varUtil.walkStorage(type, location)
+        ret.push({
+          name: variable.attributes.name,
+          type: type,
+          location: loc.currentLocation
+        })
+        location = loc.nextLocation
+      }
+    }
+    return ret
+  },
+
+  /**
+   * return state var and type definition of the given contract
+   *
+   * @param {Object} astList  - AST nodes of all the sources
+   * @param {String} contractName  - contract for which state var should be resolved
+   * @return {Object} - return the AST node of found state var
+   */
+  getStateDefinition: function (astList, contractName) {
+    var id = this.getContractId(astList, contractName)
+    if (id) {
+      var stateVar = []
+      var ctrs = this.getContractsDefinition(astList)
+      var baseContracts = this.getBaseContracts(astList, id, ctrs)
+      for (var k in baseContracts) {
+        var ctr = baseContracts[k]
+        for (var i in ctr.children) {
+          stateVar.push(ctr.children[i])
+        }
+      }
+      return stateVar
+    }
+    return null
+  },
+
+  /**
+   * return contract id given his name
+   *
+   * @param {Object} astList  - AST nodes of all the sources
+   * @param {String} contractName  - contract for which based contracts should be resolved
+   * @return {Object} - return the AST node of found contract
+   */
+  getContractId: function (astList, contractName) {
+    var id
+    this.walkAstList(astList, { 'ContractDefinition': function (node) {
+      if (node.attributes && node.attributes.name === contractName) {
+        id = node.id
+        return false
```

Comment:
I would say this is an optimization that is mostly irrelevant. We will still visit all the other contracts.